CONTRIBUTING.rst

```diff
@@ -44,21 +44,12 @@ These images are all built from our `automated Docker Hub repository`_. The
 automated build rules include pattern matching on Git tags. The current tags
 are defined in the :doc:`README`.
 
-We follow `semantic versioning`_, but drop the bug fix level version number for
-our images, as this level of granularity is not important for any application of
-these images.
+We follow `calendar versioning`_ together with the Ubuntu LTS version for that particular image.
+For example, if the Ubuntu version is 20.04 and it is released today,
+it will be ``ubuntu20-2020.08.30`` (YYYY.MM.DD)
```

Comment:
What's the reason for adding additional version information to the build image? It seems like `ubuntu20` would be enough, we're not going to support multiple `ubuntu20` images in prod are we?

Reply:
We are not going to support multiple `ubuntu20` images in prod. That's correct.

However, I was proposing to have dated images to replace the SHA512 we are currently using to pull down an exact build --which will be easier to read and understand where we are.

Another potential usage could be to swap production docker images just with a setting (see https://github.com/readthedocs/readthedocs-ops/issues/1004). So, we could have `ubuntu20-2020.08.30` and `ubuntu20-2020.09.13` in the builders and change from one to the other without needing re-deploy --and if something goes wrong with the new image, we can rollback immediately.

Nothing of this is strictly required, just a way to version our images to avoid having multiple images that are called the same, `ubuntu20`, without knowing the differences between each of them.

Note that we can start versioning the images in Docker Hub (or whatever our registry be) and git, _without_ versioning them in the builders to simplify this for now. In the future, we can implement the "deploy by setting" or other features. I think it's good to start with the versioning at the registry and git, at least.

Reply:
Hot/warm reload settings has been somewhere on a nice to have list for me, but I've never found a compelling case to complicate our settings infrastructure more here. I'd say at very least, this would be a ways down our roadmap.

A more direct identifier does maybe sound helpful, SHAs are definitely more annoying to pick out. As long as we are only referencing the date versioned SHA when pulling from docker hub, and this isn't going to be surfaced in the application at all, the additional date on the image name seems like a great plan :+1:

Reply:
> and this isn't going to be surfaced in the application at all

Yeah, we are not going to use this calver in the application. The image will be tagged as `ubuntu-20.04` on the servers.